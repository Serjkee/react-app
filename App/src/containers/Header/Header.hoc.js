import {connect} from 'react-redux'
import Header from './Header'

const mapStateToProps = (state) => ({apiData: state.movies.apiData, sortBy: state.filters.sortBy, searchValue: state.filters.searchValue, searchBy: state.filters.searchBy})

export default connect(mapStateToProps)(Header)