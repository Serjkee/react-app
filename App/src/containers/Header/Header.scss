.header {
  display: flex;
  flex-direction: column;
  max-width: 1024px;
  margin: 0 auto;
}

.header-image {
  background: linear-gradient(
    rgba(0, 0, 0, 0.8), 
    rgba(0, 0, 0, 0.45)
  ), url('../../assets/images/headerBackground.jpg');
  background-size: cover;
  background-repeat: no-repeat;
}

.logo-header {
  display: flex;
  align-items: center;
  justify-content: space-between;
}