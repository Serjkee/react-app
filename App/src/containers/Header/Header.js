import React, { Component } from 'react'
import { withRouter, Link } from 'react-router-dom'


import './Header.scss'
import FindYourMovie from '../../components/Header/FindYourMovie/FindYourMovie'
import NetflixLogo from '../../components/NetflixLogo/NetflixLogo'
import Search from '../../components/Search/Search.hoc'
import OneCardFilm from '../../components/OneFilmCard/OneFilmCard'
import SearchButton from '../../components/Search/SearchButton/SearchButton'

export class Header extends Component {
  constructor(props) {
    super(props)
  }

  render() {
    let header, headerContent;

    if( this.props.location.pathname === '/' || this.props.location.pathname === '/movies') {
      header = <NetflixLogo />;
      headerContent = <><FindYourMovie /><Search /></>;
    } else if ( this.props.match.params.id ) {
      let cardToShow = this.props.apiData.filter( card => card.id === Number(this.props.match.params.id))[0];

      header = <><NetflixLogo /><SearchButton buttonClass='backToFilter' ><Link to={{pathname: `/movies`, search: `?search=${this.props.searchValue}&searchBy=${this.props.searchBy}&sortBy=${this.props.sortBy}&sortOrder=desc&limit=20`}}>Search</Link></SearchButton></>;
      headerContent = <OneCardFilm 
        cardImgPath={cardToShow.poster_path} 
        cardTitle={cardToShow.title}
        cardGenres={cardToShow.genres.join(', ')}
        cardReleaseDate={cardToShow.release_date.slice(0, 4)}
        cardRuntime={cardToShow.runtime}
        cardFilmOverview={cardToShow.overview} />
    }

    return (
      <div className='header-image'>
        <div className='header'>
          <div className='logo-header'>
            {header}
          </div>
          {headerContent}
        </div>
      </div>
    )
  }
}

export default withRouter(Header)
