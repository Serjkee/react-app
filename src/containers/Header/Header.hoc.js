import {showingSearchFilterContent} from '../../redux/movies/movies.actions'
import {connect} from 'react-redux'
import Header from './Header'

const mapStateToProps = (state) => ({oneCardShowed: state.movies.oneCardShowed, oneCardData: state.movies.oneCardData[0]})
const mapDispatchToProps = (dispatch) => ({
  showSearchFilter() {
    dispatch(showingSearchFilterContent());
  }
})

export default connect(mapStateToProps, mapDispatchToProps)(Header)