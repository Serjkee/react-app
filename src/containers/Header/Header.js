import React, { Component } from 'react'

import './Header.scss'
import FindYourMovie from '../../components/Header/FindYourMovie/FindYourMovie'
import NetflixLogo from '../../components/NetflixLogo/NetflixLogo'
import Search from '../../components/Search/Search'

export class Header extends Component {
  render() {
    return (
      <div className='header-image'>
        <div className='header'>
          <NetflixLogo />
          <FindYourMovie />
          <Search />
        </div>
      </div>
    )
  }
}

export default Header
