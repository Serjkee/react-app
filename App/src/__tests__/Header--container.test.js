import React from 'react'
import Header from '../containers/Header/Header'

describe('Tests Header container', () => {
  it('Matching snapshot', () => {
    const container = shallow(<Header />)

    expect(container).toMatchSnapshot();
  })
})
