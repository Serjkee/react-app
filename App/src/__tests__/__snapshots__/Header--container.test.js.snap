// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests Header container Matching snapshot 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
